pkgs/by-name/co/cosh/package.nix

```diff
@@ -0,0 +1,53 @@
+{ lib
+, stdenv
+, rustPlatform
+, fetchFromGitHub
+, openssl
+, pkg-config
+, unstableGitUpdater
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "cosh";
+  version = "unstable-2024-02-23";
+  src = fetchFromGitHub {
+    repo = "cosh";
+    owner = "tomhrr";
+    rev = "e13702dc77c63dc91df305d2abf5bbdd87809781";
+    hash = "sha256-4zsTK2kAELM/A3WEob9MDt++wowFhrDJwPEtR9QZ4oU=";
+  };
+
+  # the tests try to read /etc/localtime which is unavailable under the sandbox
+  doCheck = false;
+
+  nativeBuildInputs = [ pkg-config rustPlatform.bindgenHook ];
+  buildInputs = [ openssl ];
+
+  cargoLock.lockFile = ./Cargo.lock;
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    make prefix=$out all
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    make prefix=$out install
+    runHook postInstall
+  '';
+
+  passthru.updateScript = unstableGitUpdater {};
```

Comment:
```suggestion
```

Upstream doesn't have a lockfile, so this won't work as is

Reply:
Actually, that's something that should be reported upstream, as that's against best practice and causes more work for everyone else.

Reply:
It was eventually pushed just a couple of days after. @sikmir, would you like to work more on this? I could give it a go otherwise.